### DISCUSSION TITLE
Refactoring and further optimization

### DISCUSSION THREAD

**Discussion by remz1337** [tteck/Proxmox#2794] [General]
As I'm getting more familiar with all the functions and utilities in this repo, I see a few opportunities of improvements. I would like to know your thoughts on these:

1. Avoid code duplication:
    - for example, the `msg_info` function is defined both in `build.func` and `install.func`
    - This can be avoided by creating a `common.func` file with common functions that can be loaded at the beginning of `build.func` and `install.func`
2. Use `default_settings` function inside an app's script only to superseed a base configuration
    -  Instead of copy/pasting a bunch of options that rarely changes such as   `MTU=""`, `SD=""`, `NS=""` and `MAC=""`, the entire set of options could be defined somewhere else (say a `base_settings` function in `build.func`), so by calling `default_settings` after loading the base settings, scripts would only need to specify what is specific (RAM, CPU...)
3. Use environment variables to keep track of common options used across all services/LXC

- For 2 and 3:
    - This would also facilitate using the new `APT_CACHER` and `APT_CACHER_IP` options since people would only need to set them up in the base settings/environment variables
    - Nice to have: Add a menu to consult/update those variables

**Comment by tteck** [comment]
Nope

**Comment by tteck** [comment]
I could hide a bunch of variables, gotta think about making it easier to vet the code for a particular application
